arm-visualstudio/2014-04-01-preview/swagger/Csm.json

```diff
@@ -0,0 +1,880 @@
+          }
+        }
+      }
+    },
+    "ExtensionResourcePlan": {
+      "description": "Plan data for an extension resource.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the plan.",
+          "type": "string"
+        },
+        "product": {
+          "description": "Product name.",
+          "type": "string"
+        },
+        "promotionCode": {
+          "description": "Optional: the promotion code associated with the plan.",
+          "type": "string"
+        },
+        "publisher": {
+          "description": "Name of the extension publisher.",
+          "type": "string"
+        },
+        "version": {
+          "description": "A string that uniquely identifies the plan version.",
+          "type": "string"
+        }
+      }
+    },
+    "ExtensionResourceRequest": {
+      "description": "The body of an extension resource PUT request.",
+      "type": "object",
+      "properties": {
+        "location": {
+          "description": "The Azure region of the Visual Studio account associated with this request (i.e 'southcentralus'.)",
+          "type": "string"
+        },
+        "plan": {
+          "description": "Extended information about the plan being purchased for this extension resource.",
+          "$ref": "#/definitions/ExtensionResourcePlan"
+        },
+        "properties": {
+          "description": "A dictionary of extended properties. This property is currently unused.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-client-flatten": true
+        },
+        "resourceGroupName": {
+          "description": "Optional: the resource group name.",
+          "type": "string"
+        },
+        "resourceName": {
+          "description": "Optional: the resource name.",
+          "type": "string"
+        },
+        "resourceProviderNamespace": {
+          "description": "The name of the ARM resource provider. This should currently always be set to 'Microsoft.VisualStudio'.",
+          "type": "string"
+        },
+        "resourceType": {
+          "description": "Optional: the type of the resource.",
+          "type": "string"
+        },
+        "rootResourceName": {
+          "description": "Optional: the name of the root resource.",
+          "type": "string"
+        },
+        "rootResourceType": {
+          "description": "Optional: the type of the root resource.",
+          "type": "string"
+        },
+        "subscriptionId": {
+          "description": "The Azure subscription identifier.",
+          "type": "string",
+          "format": "uuid"
+        },
+        "tags": {
+          "description": "A dictionary of user-defined tags to be stored with the extension resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
```

Comment:
You're right, these are not necessary (our RP only reads the URI values). This is was another implementation detail leaking into the documentation since these are public properties. I've changed our tooling to ensure these are removed.